packages/server/api/src/app/pieces/piece-metadata-service/hooks/piece-filtering.ts

```diff
@@ -63,3 +73,103 @@ const filterBasedOnCategories = ({
         return categories.some((item) => (p.categories ?? []).includes(item))
     })
 }
+
+
+const searchWithinActionsAndTriggersAsWell = (searchQuery: string, pieces: PieceMetadataSchema[]): PieceMetadataSchema[] => {
+    const putActionsAndTriggersInAnArray = pieces.map((piece) => {
+        const actions = Object.keys(piece.actions).map((name) => piece.actions[name])
+        const triggers = Object.keys(piece.triggers).map((name) => piece.triggers[name])
+
+        return {
+            ...piece,
+            actions,
+            triggers,
+        }
+    })
+
+    const keysInAdditionToActionsAndTriggers = [ 
+        ...defaultFilterKeys,
+        {
+            name: 'actions.displayName',
+            weight: 5,
+        },
+        {
+            name: 'actions.description',
+            weight: 5,
+        },
+        {
+            name: 'triggers.description',
+            weight: 5,
+        },
+        {
+            name: 'triggers.displayName',
+            weight: 5,
+        },
+    ]
+    const fuse = new Fuse(putActionsAndTriggersInAnArray, {
+        isCaseSensitive: false,
+        shouldSort: true,
+        keys: 
+        keysInAdditionToActionsAndTriggers,
+        threshold: 0.2,
+    })
+
+    return fuse
+        .search(searchQuery)
+        .map(({ item }) => {
+            const originalPiece = pieces.find((p) => p.id === item.id)!
+            
+            const { suggestedActionsAndTriggers } = searchForRelatedActionsAndTriggers(originalPiece, searchQuery)
+                    
+            const actions = suggestedActionsAndTriggers.reduce<Record<string, ActionBase>>((filteredActions, suggestion) => {
+                if (suggestion.key in originalPiece.actions) {
+                    filteredActions[suggestion.key] = originalPiece.actions[suggestion.key]
+                }
+                return filteredActions
+            }, {})
+
+            const triggers = suggestedActionsAndTriggers.reduce<Record<string, TriggerBase>>((filteredTriggers, suggestion) => { 
+                if (suggestion.key in originalPiece.triggers) {
+                    filteredTriggers[suggestion.key] = originalPiece.triggers[suggestion.key]
+                }
+                return filteredTriggers
+            }
+            , {})
+            return {
+                ...originalPiece,
+                actions,
+                triggers,
+            }
+            
+        })
+
+}
+function searchForRelatedActionsAndTriggers(originalPiece: PieceMetadataSchema, searchQuery: string):
+{
+    suggestedActionsAndTriggers: { description: string, displayName: string, key: string }[]
+} {
+  
+    const actionsAndTriggers = [
```

Comment:
This thing can be replaced with

TLDR if return type is one variable why return an object?
Onject.keys is same as Object.values.
Remove variable and make it inline in the end
```ts
function searchForRelatedActionsAndTriggers(originalPiece: PieceMetadataSchema, searchQuery: string): (ActionBase | TriggerBase)[] {
    const nestedFuse = new Fuse([
        ...Object.values(originalPiece.actions),
        ...Object.values(originalPiece.triggers),
    ], {
        isCaseSensitive: false,
        shouldSort: true,
        keys: ['displayName', 'description'],
        threshold: 0.2,
    })
    return nestedFuse.search(searchQuery, { limit: 3 }).map(({ item }) => item)
}

```

Reply:
I don't like inline values when they are editable, it forces me to re-read the code and find out where to edit the value, that's why I use consts, I moved it to the top, in regards for the return type, it is made on purpose, so we can map the array back to a Record<string, ActionBase | TriggerBase>